src/sizzle.js

```diff
@@ -580,6 +580,9 @@ setDocument = Sizzle.setDocument = function( node ) {
 
 	// Return early if doc is invalid or already selected
 	if ( doc === document || doc.nodeType !== 9 || !doc.documentElement ) {
+		if ( !document ) {
+			throw new Error( "Sizzle requires the document element" );
```

Comment:
I don't want to start synthesizing errors for bad environment, but I would be willing to _remove_ some of these guards and let errors happen naturally. Please try updating the early return condition to just `if ( doc === document )` and see what happens with the test suite.

Reply:
- `if ( doc === document )` - couldn't build, got errors
  ````
  PhantomJS 2.1.1 (Linux 0.0.0) selector class FAILED
	Died on test #24 global code@test/unit/selector.js:256:11: undefined is not a constructor (evaluating 'document.createElement("fieldset")')
	assert@dist/sizzle.js:387:33
	setDocument@dist/sizzle.js:612:29
	matchesSelector@dist/sizzle.js:988:14
	test/unit/selector.js:298:36
	runTest@node_modules/qunitjs/qunit/qunit.js:843:32
	run@node_modules/qunitjs/qunit/qunit.js:828:11
	node_modules/qunitjs/qunit/qunit.js:970:14
	process@node_modules/qunitjs/qunit/qunit.js:629:24
	begin@node_modules/qunitjs/qunit/qunit.js:611:9
	node_modules/qunitjs/qunit/qunit.js:671:9
	
	Expected 27 assertions, but 24 were run
	global code@test/unit/selector.js:256:11
  ...
  ````
- `if ( doc === document || doc.nodeType !== 9 )` - built, tested, no errors (it's pretty much like the original version)

> I don't want to start synthesizing errors for bad environment

\>> Then I think this PR should be closed.

---
This is not related to this PR: `test/index.html` every test fails on Firefox (`beforeEach failed on element: context is null`), Chrome wouldn't load the test (`jquery.js:8526 Access to XMLHttpRequest at 'Sizzle/test/data/fixtures.html' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, https.`) but it works on Edge - all the tests pass, is this a known issue or should I open a new issue?